frontend/lib/src/components/elements/DeckGlJsonChart/useDeckGl.tsx

```diff
@@ -96,10 +96,52 @@ const interpolate = (info: PickingInfo, body: string): string => {
   return body
 }
 
+const getShouldUseOriginalFillFunction = (
+  selectedIndices: number[],
+  anyLayersHaveSelection: boolean,
+  visualLayerSelectionMode: VisualLayerSelectionMode
+): boolean => {
+  switch (visualLayerSelectionMode) {
+    case "perLayer":
+      return selectedIndices.length === 0
+    case "perMap":
+      return !anyLayersHaveSelection
+    default:
+      assertNever(visualLayerSelectionMode)
+      return false
+  }
+}
+
 export const useDeckGl = (props: UseDeckGlProps): UseDeckGlShape => {
-  const { element, height, isLightTheme, width } = props
+  const { element, height, isLightTheme, theme, widgetMgr, width } = props
   const { tooltip, useContainerWidth: shouldUseContainerWidth } = element
 
+  //#region QueryString Parsing
```

Comment:
never seen this before, what does this do?

Reply:
oh cool! does vscode natively support this, or do you need an extension?

Reply:
It is natively supported!